### PR TITLE
Fix qute:history on QtWebKit.

### LINKED ISSUE #2472: qute:history broken with QtWebKit
With a non-NG QtWebKit, opening `qute:history` either shows nothing or (after some reloading and reopening) lots of entries with the title `undefined` and time `-8:-31:-23 GMT+1`.

It also logs this:

```
XMLHttpRequest cannot load qute://history/data. Cross origin requests are only supported for HTTP.
NETWORK_ERR: XMLHttpRequest Exception 101: A network error occurred in synchronous requests.
```

Not sure if we can do anything about this, other than adding the old code as a fallback page again and using that with QtWebKit...

We also really should've had a test catching this.

cc @imransobir

### PR DESCRIPTION
**Pull request by imransobir**
Use old history page on QtWebKit and when javascript is disabled.

Fixes #2472.

### CODE REVIEW

File: qutebrowser/browser/qutescheme.py [hunk 1]
```diff
@@ -234,21 +255,49 @@ def history_iter(start_time, reverse=False):
...
-            history = reversed(list(history_iter(start_time, reverse=False)))
-
-        return 'text/html', json.dumps(list(history))
+        return 'text/html', json.dumps(history_data(start_time))
     else:
-        return 'text/html', jinja.render('history.html', title='History',
-                session_interval=config.get('ui', 'history-session-interval'))
+        if (
+            config.get('content', 'allow-javascript') and
+            objects.backend == usertypes.Backend.QtWebEngine
```
Comment (The-Compiler) [inline qutebrowser/browser/qutescheme.py]: You can also use the new page if `objects.backend == usertypes.Backend.QtWebKit and qtutils.is_qtwebkit_ng(qWebKitVersion())` (however you'll need to do a conditional import for `qWebKitVersion()`)

File: tests/end2end/features/history.feature [hunk 2]
```diff
@@ -77,7 +77,7 @@ Feature: Page history
     Scenario: Listing history
         When I open data/numbers/3.txt
         And I open data/numbers/4.txt
-        And I open qute://history/data
+        And I open qute://history
         Then the page should contain the plaintext "3.txt"
         Then the page should contain the plaintext "4.txt"
```
Comment (The-Compiler) [inline tests/end2end/features/history.feature]: Do you have an idea why this didn't fail with the broken history page before?

Comment (imransobir) [inline tests/end2end/features/history.feature]: Probably webkit preventing JS from accessing qute:history/data, but allowing it to be accessed directly.

**Comment by The-Compiler** [comment]
I'm so confused right now... The last time I checked with QtWebKit-NG the new page seems to work fine, but when I tried again now, I got the same error suddenly. Yet the tests seem to pass with QtWebKit-NG on Travis?!

**Comment by The-Compiler** [comment]
Oh, looks like `qute://history` works, but `qute:history` displays nothing. Can we just redirec to `qute://history` when `qute:history` is loaded?

**Comment by imransobir** [comment]
Aren't qute:history and qute://history the same thing? If not, how would I go about redirecting qute:history to qute://history?

**Comment by The-Compiler** [comment]
From the URL's perspective, they're not. `qute://history` is an URL for the host `history` without a path:

```
>>> u1 = QUrl('qute://history')
>>> u1.host()
'history'
>>> u1.path()
''
```

While `qute:history` is an URL without a host and with the path `history`:

```
>>> u2 = QUrl('qute:history')
>>> u2.host()
''
>>> u2.path()
'history'
```

qutebrowser currently simply [handles both cases](https://github.com/qutebrowser/qutebrowser/blob/v0.10.x/qutebrowser/browser/qutescheme.py#L128-L134), but I'm guessing for the browser it's not the same origin. `qute://history/` and `qute://history/data` are the paths `/` and `/data` on the host `history`, while `qute:history` and `qute://history/data` are two different hosts.

Not sure how exactly to do the redirect. I wonder if we can cause a HTTP redirect inside the scheme handler. If not, we might need to use JS to do so. I'll look into it if I find the time, unless you're faster :wink:

**Comment by The-Compiler** [comment]
I got something to work, but it'll need some cleanup before pushing it.

**Comment by The-Compiler** [comment]
I merged this and implemented the redirection in 4ec5700cbf9ced71c24d57de3ddcae102c9116c0 - ended up being a bit more work than I thought it'd be, but it seems to work. Now both `qute:history` and `qute://history` work fine in all scenarios from what I can tell.

Thanks!

**Pull request merged**